src/movies.js

```diff
@@ -1,22 +1,71 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+    return moviesArray.map(movie => movie.director)
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+    let count = 0;
+    
+    moviesArray.filter (function(movie){
+        if (movie.director === "Steven Spielberg" && movie.genre.includes ("Drama")) {
+            count ++;
+        } 
+    })
+    
+    return count;
```

Comment:
mmmm filter nos vuelve un array, si nos devuelve un array podemos usar uno de sus atributos, es decir que tenemos el length!

```suggestion
    return moviesArray.filter((movie) => movie.director === "Steven Spielberg" && movie.genre.includes ("Drama")).length
```